### PR TITLE
Use Newtonsoft.Json 11 in V3

### PR DESCRIPTION
**Pull request by SzymonPobiega**
I had to add reference to json in the test projects because ApprovalTests is still on json 10.

### CODE REVIEW

File: src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj [hunk 1]
```diff
@@ -14,6 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="[11.0.1, 12.0.0)" />
```
Comment (danielmarbach) [inline src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj]: Why are these additions necessary? Plus tests would not use the range but 11.0.1 directly

File: src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="[11.0.1, 12.0.0)" />
```
Comment (danielmarbach) [inline src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj]: Why are these additions necessary? Plus tests would not use the range but 11.0.1 directly

File: src/SqlPersistence.Tests/SqlPersistence.Tests.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="[11.0.1, 12.0.0)" />
```
Comment (danielmarbach) [inline src/SqlPersistence.Tests/SqlPersistence.Tests.csproj]: Why are these additions necessary? Plus tests would not use the range but 11.0.1 directly

File: src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs [hunk 4]
```diff
@@ -32,10 +32,13 @@ TypeReference GetTypeReference(object value)
...
     public bool HasProperties => true;
+    public bool HasConstructorArguments => false;
     public Collection<CustomAttributeNamedArgument> Fields => throw new NotImplementedException();
 
     public Collection<CustomAttributeNamedArgument> Properties
     {
         get;
     }
+
+    public Collection<CustomAttributeArgument> ConstructorArguments => new Collection<CustomAttributeArgument>();
```
Comment (danielmarbach) [inline src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs]: How is this related to Newtonsoft update?

File: src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj (hunk 1 above)
Comment (danielmarbach) [reply src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj]: Ah saw it now in the PR description. But still the test should not use the range

File: src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs (hunk 4 above)
Comment (SzymonPobiega) [reply src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs]: Nope but we have a wildcard dependency on mono.cecil and they released a new breaking version.

File: src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj (hunk 1 above)
Comment (SzymonPobiega) [reply src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj]: Fixed to 11.0.1

File: src/SqlPersistence.Tests/SqlPersistence.Tests.csproj (hunk 3 above)
Comment (SzymonPobiega) [reply src/SqlPersistence.Tests/SqlPersistence.Tests.csproj]: Fixed to 11.0.1

File: src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj (hunk 2 above)
Comment (SzymonPobiega) [reply src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj]: Fixed to 11.0.1

File: src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs (hunk 4 above)
Comment (danielmarbach) [reply src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs]: Would it make sense to lock in the version?

Comment (SzymonPobiega) [reply src/ScriptBuilder.Tests/CecilExtensions/CustomAttributeMock.cs]: I am not sure. We didn't have problems with this dependency so far.
